offer-manager-domain/src/main/groovy/life/qbic/business/projects/create/CreateProjectDataSource.groovy

```diff
@@ -0,0 +1,34 @@
+package life.qbic.business.projects.create
+
+import life.qbic.business.exceptions.ApplicationDeniedException
+import life.qbic.business.exceptions.DatabaseQueryException
+import life.qbic.datamodel.dtos.projectmanagement.Project
+import life.qbic.datamodel.dtos.projectmanagement.ProjectApplication
+
+/**
+ * Provides access to the project management datasource
+ *
+ * This interface collects methods to interact with the project management
+ * datasource in the context of the Create Project use case.
+ *
+ * @since 1.0.0
+ */
+interface CreateProjectDataSource {
+
+    /**
+     * Creates a new QBiC project in the data management platform.
+     *
+     * @param projectApplication A project application with the information necessary for the
+     * project registration
+     *
+     * @return Information about the created project
+     *
+     * @since 1.0.0
+     * @throws ApplicationDeniedException If the application was denied. Reasons for denial are
+     * currently:
+     *  1. A project with the same project title already exists
```

Comment:
I think it is good that you brought this up. However, I think it would make sense to forward this to the output instead of throwing an exception. This would make it more declarative.

You could extend the output interface with a method `applicationDenied(projectApplication:ProjectApplication):void` 
Then for every other reason the use case would choose to neither fail nor succeed (e.g. have a defined return point) you could extend the interface. For example, if you were to decide that you only allow for certain project managers you could add a method `unauthorizedProjectManager(projectManager:ProjectManager):void` etc.

Reply:
Oh sorry thought this was the input interface. Should the business logic for denial be handled by the database? I would think the responsibility lies with the use case.

Reply:
Yes, good point. I think i will refactor it to provide a method in the output interface.

However I want to have another Exception thrown than the DataBaseQueryException, because this is very generic. Especially if a project already exists. I will suggest another name for this type of Exception to be thrown by the data source.